Framework/PythonInterface/plugins/algorithms/ANSTO/emu_common.py

```diff
@@ -0,0 +1,254 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import math
+import numpy as np
+
+from typing import Dict, Any
+
+from mantid import mtd
+
+from ansto_common import (
+    HdfKey,
+    extract_hdf_params,
+    FilterPixelsTubes,
+    IniParameters,
+    IntOption,
+    StrOption,
+    RangeOption,
+)
+
+
+# the following components are shared by the inelastic and elastic EMU reduction algorithms
+class DopplerSupport:
+    _frequency: float
+    _amplitude: float
+    _speed: float
+    _phase: float
+    _frequency_option: str
+    _phase_option: str
+    _table: str
+
+    def __init__(self, frequency_option: str, phase_option: str, table: str) -> None:
+        self._frequency_option = frequency_option
+        self._phase_option = phase_option
+        self._table = table
+
+    @property
+    def frequency(self) -> float:
+        return self._frequency
+
+    @property
+    def amplitude(self) -> float:
+        return self._amplitude
+
+    @property
+    def speed(self) -> float:
+        return self._speed
+
+    @property
+    def phase(self) -> float:
+        return self._phase
+
+    def set_amplitude_speed(self, sample_file: str, ds_index: int = 0) -> None:
+        tags = [
+            HdfKey("dopp_amp", "/entry1/instrument/doppler/ctrl/amplitude", None),
+            HdfKey("dopp_vel", "/entry1/instrument/doppler/ctrl/velocity", None),
+        ]
+        values, _ = extract_hdf_params(sample_file, tags)
+        # mm to metres
+        self._amplitude = float(values["dopp_amp"][ds_index]) * 0.001
+        self._speed = float(values["dopp_vel"][ds_index])
+        self._complete_doppler_params()
+
+    def check_doppler_settings(self, ws_tag: str) -> None:
+        # confirm that the doppler amplitude and speed match that of the sample
+        mrun = mtd[ws_tag].getRun()
+        amp = mrun.getProperty("DopplerAmplitude").value[0]
+        spd = mrun.getProperty("DopplerVelocity").value[0]
+        if math.fabs(self._amplitude - amp) > 0.001 or math.fabs(self._speed - spd) > 0.1:
```

Comment:
Might be an idea to define the tolerances as global constants at the top of the file to make it easier to change (and if you then decide to use it in different places...)?

Reply:
Done.